src/EditorFeatures/Core/Shared/Utilities/WorkspaceThreadingServiceProvider.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Shared.Utilities
+{
+    [ExportWorkspaceService(typeof(IWorkspaceThreadingServiceProvider)), Shared]
+    internal sealed class WorkspaceThreadingServiceProvider : IWorkspaceThreadingServiceProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public WorkspaceThreadingServiceProvider(
+            IWorkspaceThreadingService service)
+        {
+            Service = service;
+        }
+
+        public IWorkspaceThreadingService Service { get; }
```

Comment:
@jasonmalinowski i needed this for teh DocumentNavigationOperation change below.  That's a public API we can't pass an IThreadingContext into.  INstead, it just gets a WOrkspace, so we need a way to get this from that.

Reply:
Yeah, this seems reasonable in the sense that I don't have a good alternative here.